modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchHandler.kt

```diff
@@ -157,7 +169,7 @@ class SearchHandler @Inject constructor(
             val searchPodcastsResult = (localPodcastsResult + serverPodcastsResult).distinctBy { it.uuid }
             val searchEpisodesResult = serverSearchResults.episodeSearch.episodes
 
-            if (serverSearchResults.searchTerm.isEmpty() || searchPodcastsResult.isNotEmpty() || serverSearchResults.error != null) {
+            if (serverSearchResults.searchTerm.isEmpty() || (searchPodcastsResult.isNotEmpty() || searchEpisodesResult.isNotEmpty()) || serverSearchResults.error != null) {
```

Comment:
Do you think it would be worth extracting out at variable with some explanation to make this easier to follow?

I came up with this based on my (possibly incorrect) understanding of what we're doing here, but I'm sure it could be improved.

``` kotlin
val hasResults =
    serverSearchResults.searchTerm.isEmpty() // if the search term is empty, we don't have "no results"
        || (searchPodcastsResult.isNotEmpty() || searchEpisodesResult.isNotEmpty()) // check if there are any results
        || serverSearchResults.error != null // an error is a result
```

 Feel free to leave as-is if you prefer.

Reply:
Your comments make sense to me so I added them in 3ef4879.